### DISCUSSION TITLE
Change detection when mixing queries and resources

### DISCUSSION THREAD

**Discussion by nsabovic** [bevyengine/bevy#5888] [Q&A] [answered]
Is there a way to run a system only when a resource changes but still access the entities?

Let's say I have a resource which specifies some property and a bunch of entities which need to update when that property changes. If I write something like this...

```
fn my_system(
    property: Res<Property>,
    mut query: Query<(&mut MyComponent)>,
)
```

...I get called each frame. I would ideally want to get called only when `Property` changes, but I do want to be able to mutate the entities in response to it.

Am I doing this wrong?

Thanks!

**Comment by nicopap** [comment, marked as answer]
You would be using the `Res::is_changed` method

```rust
fn my_system(
    property: Res<Property>,
    mut query: Query<(&mut MyComponent)>,
) {
  if !property.is_changed() {
    return;
  }
  // Update your `MyComponent`s
}
```

Systems do not have a huge overhead, so it's not a big deal to let it run and check that your resource hasn't changed.

**Reply by nsabovic** [reply]
Perfect! Thank you.
